recursion_exercise/recursion.js

```diff
@@ -0,0 +1,56 @@
+function productOfArray(arr) {
+    if(arr.length === 1) return arr[0]
+    return arr[0] * productOfArray(arr.slice(1))
+}
+
+
+function collectStrings(obj) {
+    var newArr = [];
+
+    function collectStringsHelper(obj) {
+        for(var i=0; i<Object.keys(obj).length; i++) {
```

Comment:
when looping over an object, always try to use a `for in` loop just like you have with the `contains` function.